java/vector/src/main/java/org/apache/arrow/vector/ipc/message/MessageReaderBase.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.vector.ipc.message;
+
+import java.io.IOException;
+
+import org.apache.arrow.vector.ipc.ReadChannel;
+
+/**
+ * Abstract class to define a reader for a sequence of messages using a ReadChannel. Type parameter
+ * T will be used by the implementing class to hold the message body.
+ */
+public abstract class MessageReaderBase<T> implements AutoCloseable {
```

Comment:
Message Reader should be built for consumers, not MessageReader implementers. Also, it shouldn't require the use of a ReadChannel. Part of the reason that I jumped through hoops in the flight patch is that the reader base assumes you have a readchannel. That's an implementation detail of a particular reader implementation